File: re-ssim-components/re-ssim-data-kingo-application/src/main/java/com/shinonometn/re/ssim/data/kingo/application/caterpillar/base/calendar/SchoolCalendar.java
package com.shinonometn.re.ssim.data.kingo.application.caterpillar.base.calendar;

import com.shinonometn.re.ssim.service.caterpillar.common.SchoolDate;

import java.time.LocalDateTime;

public interface SchoolCalendar {
    public String getName();

    public LocalDateTime getStartDate();

    LocalDateTime getEndDate();

    public int getDaysOfTerm();

    public SchoolDate getFromDateTime(LocalDateTime dateTime);
}

File: re-ssim-caterpillar/re-ssim-caterpillar-core/src/main/java/com/shinonometn/re/ssim/caterpillar/core/calendar/SchoolCalendar.kt
package com.shinonometn.re.ssim.caterpillar.core.calendar

import java.time.LocalDateTime

interface SchoolCalendar {

    val name: String
    val startDate: LocalDateTime
    val endDate: LocalDateTime
    val daysOfTerm: Int

    fun getFromDateTime(dateTime: LocalDateTime): SchoolDate
}
